src/screens/main/home/index.js

```diff
@@ -0,0 +1,79 @@
+import React, {useState, useEffect} from 'react';
+import {ScrollView} from 'react-native';
+import LinearGradient from 'react-native-linear-gradient';
+import Styles from './Styles';
+import CategoryCard from '../../../components/categorycard';
+
+export default Home = ({navigation}) => {
+  const [topRatedMovies, setTopRatedMovies] = useState([]);
+  const [popularMovies, setPopularMovies] = useState([]);
+  const [upComingMovies, setUpComingMovies] = useState([]);
+
+  useEffect(() => {
+    getTopRatedMovies();
+  }, []);
+  useEffect(() => {
+    getPopularMovies();
+  }, []);
+  useEffect(() => {
+    getUpComingMovies();
+  }, []);
+
+  const getTopRatedMovies = async () => {
+    await fetch(
+      'https://api.themoviedb.org/3/movie/top_rated?api_key=8ca2abb7154c8c81ef7cb403c11eec32&language=en-US&page=1',
+    )
+      .then((response) => response.json())
+      .then((data) => {
+        setTopRatedMovies(data.results);
+      })
+      .catch((err) => alert(err));
+  };
+  const getPopularMovies = async () => {
+    await fetch(
+      'https://api.themoviedb.org/3/movie/popular?api_key=8ca2abb7154c8c81ef7cb403c11eec32&language=en-US&page=1',
```

Comment:
make base URL and api key as constants and pass other variables dynamically.

Reply:
You can also create a method that would take the dynamic parameters, make the request call and return the response to the calling method.